ogorek.go

```diff
@@ -448,8 +458,19 @@ type Call struct {
 }
 
 func (d *Decoder) reduce() error {
-	args := d.pop().([]interface{})
-	class := d.pop().(Class)
+	if len(d.stack) < 2 {
```

Comment:
I think in these cases I would rather attempt the `pop` and then return the error if there is one. It makes better use of the returned `pop` error (and what if there is some other error it can return?) as well as absolves each function from having to count how many times it will call pop.